packages/wallet/src/wallet.ts

```diff
@@ -1,28 +1,25 @@
 import { AdvanceRequestHandler, Voucher } from "@deroll/app";
-import {
-    Address,
-    encodeFunctionData,
-    erc20Abi,
-    getAddress,
-    isAddress,
-} from "viem";
-
-import { cartesiDAppAbi, dAppAddressRelayAddress } from "./rollups";
-import {
-    isERC20Deposit,
-    isEtherDeposit,
-    parseERC20Deposit,
-    parseEtherDeposit,
-} from ".";
+import { Address, getAddress, isAddress } from "viem";
+
+import { inspect } from "node:util";
+import { TokenHandler } from "./token";
 
 export type Wallet = {
     ether: bigint;
-    erc20: Record<Address, bigint>;
+    erc20: Map<Address, bigint>;
+    erc721: Map<Address, Set<bigint>>;
+    erc1155: Map<Address, Map<bigint, bigint>>;
 };
 
 export interface WalletApp {
     balanceOf(address: string): bigint;
     balanceOf(token: Address, address: string): bigint;
```

Comment:
Maybe rename to `balanceOfERC20` for consistency

Reply:
`balanceof` is still using the current method that verifies the Ether or ERC20 balance.I can change too.